bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java

```diff
@@ -0,0 +1,763 @@
+        connectRetryJob = scheduler.schedule(this::connect, waitMinutes, TimeUnit.MINUTES);
+    }
+
+    private synchronized void disconnect() {
+        logger.debug("Disconnecting");
+
+        Thread senderThread = this.senderThread;
+        Thread readerThread = this.readerThread;
+
+        if (connectRetryJob != null) {
+            connectRetryJob.cancel(true);
+        }
+        if (keepAlive != null) {
+            keepAlive.cancel(true);
+        }
+
+        // May be called from keepAliveReconnect thread, so call cancel with false
+        reconnectTaskCancel(false);
+
+        if (senderThread != null && senderThread.isAlive()) {
+            senderThread.interrupt();
+        }
+        if (readerThread != null && readerThread.isAlive()) {
+            readerThread.interrupt();
+        }
+        if (sslsocket != null) {
+            try {
+                sslsocket.close();
+            } catch (IOException e) {
+                logger.debug("Error closing SSL socket: {}", e.getMessage());
+            }
+            sslsocket = null;
+        }
+        if (reader != null) {
+            try {
+                reader.close();
+            } catch (IOException e) {
+                logger.debug("Error closing reader: {}", e.getMessage());
+            }
+        }
+        if (writer != null) {
+            try {
+                writer.close();
+            } catch (IOException e) {
+                logger.debug("Error closing writer: {}", e.getMessage());
+            }
+        }
+
+        deviceDataLoaded.set(false);
+        buttonDataLoaded.set(false);
+    }
+
+    private synchronized void reconnect() {
+        logger.debug("Attempting to reconnect to the bridge");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "reconnecting");
+        disconnect();
+        connect();
+    }
+
+    /**
+     * Method executed by the message sender thread (senderThread)
+     */
+    private void senderThreadJob() {
+        logger.debug("Command sender thread started");
+        try {
+            while (!Thread.currentThread().isInterrupted() && writer != null) {
+                LeapCommand command = sendQueue.take();
+                logger.trace("Sending command {}", command);
+
+                try {
+                    BufferedWriter writer = this.writer;
+                    if (writer != null) {
+                        writer.write(command.toString() + "\n");
+                        writer.flush();
+                    }
+                } catch (IOException e) {
+                    logger.warn("Communication error, will try to reconnect. Error: {}", e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    sendQueue.add(command); // Requeue command
+                    reconnect();
+                    break; // reconnect() will start a new thread; terminate this one
+                }
```

Comment:
You should made another catch for a InterruptedIOException. You shouldn't attempt a reconnect if you received an interrupt.

Reply:
When catching InterruptedIOException, is it best to call Thread.currentThread().interrupt() or just exit?

Reply:
It would be best to just exit. Calling `Thread.currentThread().interrupt()` is only useful if you expect/know that something higher up the call stack is checking for interrupts. Which I don't think is the case here.

Reply:
Done